src/components/Modal/Modal.cy.tsx

```diff
@@ -1,20 +1,39 @@
 import React from "react";
-import Modal from "@/components/Modal/Modal";
+import Modal, { ModalProps } from "@/components/Modal/Modal";
+import StyleManager from "@/app/themes";
+
+const StyledModal: React.FC<ModalProps> = (props) => {
```

Comment:
Oh the reason that this is now needed is because for cypress component test, it is not wrapped under StyleManager, but the component itself assumes it is wrapped and the styled component will start trying to fetch things from props.theme.primary.foreground[0] etcetc.

Before my current implementation, it is not necessary because the color is just index as props.theme.primaryColor for instance, so theme.primaryForegroundColor is undefined so the color is just set as undefined

But for the current implementation, since props.theme.primary is undefined, then I can't use .foreground[0] on it as undefined as no property foreground, then it will throw me an error. Tom also doesn't recommend me to use props.theme.primary?.foreground[0], so it has to be wrapped under style manager